docs/docs/design/collator-protocol-validator-side.md

```diff
@@ -0,0 +1,134 @@
+---
+layout: default
+title: Collator Protocol - Validator Side
+permalink: /design/collator-protocol-validator-side/
+---
+
+# Collator Protocol: Validator Side
+
+The Collator Protocol implements the network protocol by which collators and validators communicate. It is used by collators to distribute collations to validators and used by validators to accept collations by collators.
+
+The **Validator-Side**, as any subsystem, has an inner state and a `Run` method with a action loop. However differently from other subsystems, the **Validator-Side** does not expect handling `ActiveLeavesSignal`.
+
+```mermaid
+graph TD
+    A[run] --> B[run_inner]
+    B --> C{select! loop}
+    
+    C -->|reputation_delay| D[send reputation]
+    C -->|ctx.recv| E[process_msg]
+    C -->|next_inactivity_stream| F[disconnect_inactive_peers]
+    C -->|collation_requests| G[handle_collation_fetch_response]
+    C -->|collation_fetch_timeouts| H[dequeue_next_collation_and_fetch]
+    
+    E --> I[handle_network_msg]
+    E --> J[handle_our_view_change]
+    
+    G -->|Success| K[kick_off_seconding]
+    G -->|Error| H
+    
+    K -->|Error| H
+    K -->|Success false| H
+    
+    H --> L[get_next_collation_to_fetch]
+    H --> M[fetch_collation]
+    
+    M --> N[request_collation]
+    
+    I --> O[process_incoming_peer_message]
+    O --> P[handle_advertisement]
+    P --> Q[enqueue_collation]
+    
+    J --> R[construct_per_relay_parent]
+    
+    K --> S[request_persisted_validation_data]
+    K --> T[request_prospective_validation_data]
+    K --> U[second_unblocked_collations]
+```
+
+## State
+
+The subsystem state is composed by:
+
+- Active Leaves
+- Collator's data
+- Current parachains we're assigned to
+- Collation request information (ongoing requests, cancelations and timeouts)
+- Collations sucessfully fetched waiting validation
+- Collations blocked (due to missing parent block data)
+
+## Action Loop
+
+The subsystem listen to a set of channels on his action loop:
+
+- Reputation Delay: used to batch reputations and send every X amount of time to Network Bridge, that is done through `Reputation Aggregator`
```

Comment:
it looks like this Reputation Aggregator is used in multiple places, but did we implement it?